test/start-kubernetes.sh

```diff
@@ -101,45 +100,41 @@ EOF
                         exit 2
                     fi
                 fi
-                unxz ${CLOUD_IMAGE}
+                unxz ${CLOUD_IMAGE} || die "failed to unpack ${CLOUD_IMAGE}"
                 CLOUD_IMAGE=${CLOUD_IMAGE/.xz}
                 ;;
             ubuntu)
                 base_url="https://cloud-images.ubuntu.com/disco/current/"
                 sha_file="SHA256SUMS"
-                curl -s -O ${base_url}/${CLOUD_IMAGE}.xz
-                curl -s -O ${base_url}/${sha_file}
-                if sha256sum $sha_file ${CLOUD_IMAGE}.xz; then
-                    unxz ${CLOUD_IMAGE}.xz
-                    CLOUD_IMAGE=${CLOUD_IMAGE/.xz}
-                fi
+                curl -s -O ${base_url}/${CLOUD_IMAGE}.xz || die "failed to download ${base_url}/${CLOUD_IMAGE}.xz"
+                curl -s -O ${base_url}/${sha_file} || die "failed to download ${base_url}/${sha_file}"
+                sha256sum $sha_file ${CLOUD_IMAGE}.xz || die "validation of ${CLOUD_IMAGE}.xz failed"
+                unxz ${CLOUD_IMAGE}.xz || die "failed to unpack ${CLOUD_IMAGE}.xz"
+                CLOUD_IMAGE=${CLOUD_IMAGE/.xz}
```

Comment:
if the idea is to append "/.xz", then this does not work. should be: CLOUD_IMAGE=${CLOUD_IMAGE}/.xz
(similar few lines below)

Reply:
hm no it probably wasnt the idea.

Reply:
but still confusing, / is substitution in bash, so we want to delete .xz  ? but why it is added just few lines above then ?

Reply:
No, that's bash syntax for *removing* the suffix.

Reply:
`CLOUD_IMAGE` starts with `.xz` suffix (for example, `CLOUD_IMAGE="clear-$TEST_DISTRO_VERSION-cloud.img.xz"`) because the full file name has to be used during downloading. Then the file gets unpacked and the suffix removed. Could have been done differently, but this approach was already used before my changes, so I have kept it. :man_shrugging:

Reply:
The logic does not add up in ubuntu, fedora cases (which are different from clear case).
1. If CLOUD_IMAGE has .xz suffix in these blocks, there will double suffix used in most of lines which is weird, and last line takes away suffix.
2. If CLOUD_IMAGE is without suffix, most of lines make sense by adding suffix, but last line is NOP as there was no suffix.
So the whole block of ubuntu or fedora can not be correct IMHO, or do you see otherwise? 
What does commands trace show when this code runs?
I believe it is the 2nd case and we have NOP stmt at the end, so its not causing error, but code should not be confusing nevertheless.

Reply:
Ah, now I see what you mean. Yes, both code blocks are currently dead code and broken. Should I remove them instead of doing partial fixes?

At least Fedora already got fixed in my "fedora" branch:
https://github.com/pohly/pmem-CSI/blob/d1d6a8a901674bc30cb101748077c6bf90515f7a/test/start-kubernetes.sh#L94-L140

Reply:
yes, removed, or marked "under construction" is more clear

Reply:
I removed it.